### ISSUE TITLE
[Bug]: Upload existing files could result in upload page freezing

### ISSUE THREAD

**Issue by scriptsagency** [nextcloud/server#37419]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

This commit https://github.com/nextcloud/server/commit/6c4d4d8bc4abc3f1b3d794da9d90906ccac12607 added a deffer in apps/files/js/file-upload.js:696 

The flow of the deffer is: when a file is removed from upload list 
"removeUpload: function(upload) { " the file will be actually removed only after 5s 
In the meanwhile, if the files already exists a promt with conflict will appear and if user selects to upload new versions or both then the methods will be called 
"onReplace:function(upload) {" OR "onAutorename:function(upload) {"
Both methods will run "this.submitUploads([upload]);" which will add back the upload to the submit list. 
BUT: depending on the time of the confirm and upload speed, the deletion of the upload could still be defferred when the upload is readded to the list resulting in the Replace or Autorename upload to actually be removed which is turn can lead to an upload freeze.

Proposed solutions: 
1. in OnReplace and onAutorename change the upload.id and upload.data.uploadId by adding an extra "_rename" or "_autoRename", in this way the upload won't be removed from the submit list by the deferred deletion which is using the upload.data.uploadId to remove it from the _uploads "delete self._uploads[uploadId];"
2. Remove the deferred removal when calling RemoveUpload since this update introduced a major issue. 

From my point of view this is a major blocking error present there which took me a lot of time to debug and understand and it is currently affecting all Nextcloud instances. I'd go with first option considering it more elegant and allowing to keep the defer in place. 

### Steps to reproduce

1. Upload in files a list of files and folders. I created a list of 10 folders with one file in each 
2. Try to upload again the list and overwrite. Wait a little after the conflict promt appears and only then select to overwrite the new version
3. In several cases the upload bar X button remains blocked and in console the following message appears 
merged-index.js?v=e5c8a9e5-0:2872 Uncaught TypeError: Cannot read properties of null (reading 'done')
    at HTMLInputElement.<anonymous> (merged-index.js?v=e5c8a9e5-0:2872:13)
    at HTMLInputElement.dispatch (jquery.js:5430:27)
    at m.handle (jquery.js:5234:28)
    at Object.trigger (jquery.js:8719:12)
    at HTMLInputElement.<anonymous> (jquery.js:8797:17)
    at Function.each (jquery.js:385:19)
    at t.<computed>.each (jquery.js:207:17)
    at t.<computed>.trigger (jquery.js:8796:15)
    at e.<computed>.<computed>._trigger (jquery-ui.js:715:16)
    at e.<computed>.<computed>._onDone (merged-index.js?v=e5c8a9e5-0:10119:18)

### Expected behavior

Upload files that already exists should work file. I tried to explain the exact problem in the code. 

### Installation method

Community Docker image

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.1

### Web server

Apache (supported)

### Database engine version

MySQL

### Is this bug present after an update or on a fresh install?

Fresh Nextcloud Server install

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [X] Other

### Configuration report

```shell
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost:8025"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "25.0.4.1",
        "overwrite.cli.url": "http:\/\/localhost:8025",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "app_install_overwrite": [
            "user_cas"
        ],
        "check_data_directory_permissions": false,
        "overwriteprotocol": "http"
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 2.17.0
  - circles: 25.0.0
  - cloud_federation_api: 1.8.0
  - comments: 1.15.0
  - contactsinteraction: 1.6.0
  - dashboard: 7.5.0
  - dav: 1.24.0
  - federatedfilesharing: 1.15.0
  - federation: 1.15.0
  - files: 1.20.1
  - files_pdfviewer: 2.6.0
  - files_rightclick: 1.4.0
  - files_sharing: 1.17.0
  - files_trashbin: 1.15.0
  - files_versions: 1.18.0
  - firstrunwizard: 2.14.0
  - logreader: 2.10.0
  - lookup_server_connector: 1.13.0
  - nextcloud_announcements: 1.14.0
  - notifications: 2.13.1
  - oauth2: 1.13.0
  - password_policy: 1.15.0
  - photos: 2.0.1
  - privacy: 1.9.0
  - provisioning_api: 1.15.0
  - recommendations: 1.4.0
  - related_resources: 1.0.4
  - serverinfo: 1.15.0
  - settings: 1.7.0
  - sharebymail: 1.15.0
  - support: 1.8.0
  - survey_client: 1.13.0
  - systemtags: 1.15.0
  - text: 3.6.0
  - theming: 2.0.1
  - twofactor_backupcodes: 1.14.0
  - updatenotification: 1.15.0
  - user_cas: 1.10.0
  - user_status: 1.5.0
  - viewer: 1.9.0
  - weather_status: 1.5.0
  - workflowengine: 2.7.0
Disabled:
  - admin_audit
  - bruteforcesettings
  - encryption
  - files_external
  - suspicious_login
  - twofactor_totp
  - user_ldap
```

**Comment by solracsf** [comment]
Cc @PVince81
